src/connectors/transformationBetween.js

```diff
@@ -0,0 +1,64 @@
+const { mat4, plane, vec2, vec3 } = require('../math/')
+
+const OrthoNormalBasis = require('../math/OrthoNormalBasis')
+
+const transform = require('./transform')
+
+/**
+ * Get the transformation matrix that connects the given connectors.
+ * @param {Object} options
+ * @param {Boolean} options.mirror=false - the 'axis' vectors should point in the same direction
+ *  true: the 'axis' vectors should point in opposite direction
+ * @param {Number} options.normalRotation=0 : the angle (RADIANS) of rotation between the 'normal' vectors
+ * @param {connector} from - connector from which to connect
+ * @param {connector} to - connector to which to connected
+ * @returns {mat4} - the matrix that transforms (connects) one connector to another
+ */
+const transformationBetween = (options, from, to) => {
+  const defaults = {
+    mirror: false,
+    normalRotation: 0
+  }
+  // mirror = !!mirror
+  const { mirror, normalRotation } = Object.assign({}, defaults, options)
+
+  // shift to the 0,0 origin
+  let matrix = mat4.fromTranslation(vec3.negate(from.point))
+
+  // align the axis
+  let axesplane = plane.fromPointsRandom(vec3.create(), from.axis, to.axis)
+  let axesbasis = new OrthoNormalBasis(axesplane)
+
+  let angle1 = vec2.angleRadians(axesbasis.to2D(from.axis))
```

Comment:
so we have multiple things inside orthoBasis:
- to2D
- getProjectionMatrix
- getInverseProjectionMatrix

There was a proposed (but closed) PR with some alternatives to all this ?

Reply:
another note: a projection matrix, is just another matrix (as in a set of operations on mat4 s) so perhaps we can do something in that direction: just adding new functions to mat4 ?

Reply:
Added Issue #184 to cleanup this mess on the next pass.